mvp-spec/partials-updater/src/mermaid.ts

```diff
@@ -0,0 +1,46 @@
+import {getAssetPath, getAssetPathRelativeToDocuments, getBinPath, getPartialPath} from "./files";
+import child from "child_process";
+import path from "path";
+
+const mmdcPath = getBinPath('mmdc')
+
+function getImageName(partialMermaidFile: string) {
+    const fileName = partialMermaidFile.substring(0, partialMermaidFile.lastIndexOf('.'));
+    return `generated-${fileName}.svg`;
+}
+
+function buildMermaidCommand(partialMermaidFile: string): string {
+    const srcPath = getPartialPath(partialMermaidFile);
+    const destFile = getImageName(partialMermaidFile);
+    const destPath = getAssetPath(destFile);
+    return `"${mmdcPath}" -i "${srcPath}" -o "${destPath}"`;
+}
+
+export async function updateMermaids(mermaidFiles: string[]): Promise<void> {
+    if (mermaidFiles.length > 0) {
+        console.log('Re-generate SVGs for Mermaid files:')
+
+        const fileNames = mermaidFiles.map(f => path.basename(f));
+
+        await Promise.all(fileNames.map(async f => {
+            const cmd = buildMermaidCommand(f);
+            await child.exec(cmd)
+            console.log(`\t${f} -> ${getImageName(f)}`)
+        }));
+
+        console.log();
+    }
+
+}
+
+export function transformMermaids(files: string[], lineBreak: string): string[] {
```

Comment:
The word "Transform" may need an alternative here. We don't transform mermaids here, we generate markdown links to images.

Reply:
Agreed, however this maps the "transformation" keyword from the config, so would you suggest we rename that too?

Reply:
If I understand well, we can see the transformation at two levels:
1. the high level, the transformation as an overall process which map with the "transformation" keyword in the config.
2. the low level, the transformation as a technical operation (i.e f: (mermaid_data) -> SVG_data)

If you take a high-level point of view, I guess the mapping that you mention can be viewed differently because this function is a sub-element of the transformation process.

So I suggest to keep the "transformation" keyword in the config and change the name of this function.

To be as much constructive as possible, here are suggestions for the name backed with a rational:

The arranged signature of the function is the following:
`f : (OrderedList[MermaidFileName]) -> OrderedList[MarkdownRefForMermaidImage]` 

So I guess this can be a good match:
`export function getMarkdownReferencesForMermaids(files: string[], lineBreak: string): string[]`
or
`export function getMarkdownRefsForMermaids(files: string[], lineBreak: string): string[]`
or 
`export function getMarkdownImageLinkForMermaids(files: string[], lineBreak: string): string[]`